src/System.Buffers.Primitives/System/Buffers/OutputWriter.cs

```diff
@@ -35,79 +35,42 @@ public void Advance(int count)
             _output.Advance(count);
         }
 
-        public void Write(byte[] source)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void Write(Span<byte> source)
```

Comment:
Why not use `ReadOnlySpan<byte>`? I think that `Write` is not supposed to write into the `source`